scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TerminationData.java

```diff
@@ -309,9 +310,23 @@ public VariablesMap getStringSerializableMap(SchedulingService service, TaskTerm
         } else {
             variablesMap.setInheritedMap(fillMapWithTaskResult(taskResult, true));
         }
+        variablesMap.setScopeMap(removeInheritedVarsFromScope(variablesMap, taskData.getTask().getVariables()));
```

Comment:
it looks like the following line, appearing earlier in the code: variablesMap.setScopeMap(taskData.getTask().getScopeVariables());
is overwritten by your code, so it should be removed

Reply:
fixed @fviale !